### PR TITLE
add interactive mode without using dialog

### PR DESCRIPTION
**Pull request by kovetskiy**
Hi. I've added support for interactive mode without using dialog, I found simple text much more pleasant than that in-terminal UI interface.

Also, in Arch Linux distribution the dialog package was in "optdepends" but actually netctl was not able to work without the dialog package. Now dialog will be really optional.

### CODE REVIEW

**Comment by joukewitteveen** [comment]
Sounds like a good idea, but I have a different suggestion.
How about we write a shim for dialog that implements just what `wifi-menu` uses and distribute it with netctl? We can then simply fallback to this shim in `wifi-menu` if dialog is not installed.

If you are interested, I can create an out-of-tree repository where we could develop it together. Any suggestions for a name?
- `shdialog` would keep in spirit with `xdialog`
- `bialog` would simply flip the first letter and point to bash
- `dialog` is also possible (it would live in `/usr/lib/netctl/dialog`)
- ?

**Comment by kovetskiy** [comment]
@joukewitteveen well, it could be implemented as a separated program, but it would just introduce new entity that we don't need right now ([Occam's razor](https://en.wikipedia.org/wiki/Occam%27s_razor))
- will it make netctl's code easier to read? I doubt it, it will introduce few more complexities to handle corner cases (like we don't need `clear` for shdialog);
- will it be easier to install netctl? No, you need new dependency;
- shdialog will be re-usable, but we don't need it right now and it will not solve any problem;
- since we introduce new program "shdialog", we need tests coverage. Yes, I have the whole ecosystem for that [http://github.com/reconquest/tests.sh](http://github.com/reconquest/tests.sh) and [https://github.com/reconquest/import.bash](https://github.com/reconquest/import.bash), but I would claim — bash is like a pain in the ass when you want to write tests.

Basically, in my personal opinion, I don't need "dialog" at all, there was only one program in my OS that used dialog — wifi-menu.

**Comment by joukewitteveen** [comment]
It need not introduce more code than this patch does. By keeping it out of `wifi-menu` it definitely makes the netctl code easier to read and maintain. Testing is also easier, because you have a proper test surface (the dialog interface).

The idea is to have a version of your compatibility code live in a separate file (`/usr/lib/netctl/shdialog`) and have `wifi-menu` fall-back to it in case dialog is not installed. The reason for developing it out of tree would be to make it easier to co-author the code as I could just add you as a collaborator.

**Comment by kovetskiy** [comment]
Hi, I still don't find any reason to move this patch to a different repository.

**Comment by joukewitteveen** [comment]
I will not pull this PR as-is. What I am willing to do is develop a shim for the dialog API together with you and then incorporate that.

**Comment by kovetskiy** [comment]
Good for you.

**Comment by joukewitteveen** [comment]
I take it that means you are not interested in pursuing this functionality any further?

**Comment by kovetskiy** [comment]
As I said above, I don't see any reasons to move it to yet another repository. 

Sorry, but I'm not interested in maintaining pointless program which will be simple wrapper around `read` bash builtin.

**Comment by joukewitteveen** [comment]
Good for you.

**Comment by kovetskiy** [comment]
Yep.
